tools/test_sbank.sh

```diff
@@ -0,0 +1,20 @@
+set -e
+# Run sbank to generate XML and HDF banks
+sbank --approximant IMRPhenomD --aligned-spin --mass1-min 15.0 --mass1-max 25.0 --spin1-min 0.0 --spin1-max 0.5 --match-min 0.97 --flow 20.0 --noise-model aLIGOZeroDetHighPower --output-filename BBH-IMRPhenomD-aLIGOZeroDetHighPower.xml --convergence-threshold 25
+
+sbank --approximant IMRPhenomD --aligned-spin --mass1-min 15.0 --mass1-max 25.0 --spin1-min 0.0 --spin1-max 0.5 --match-min 0.97 --flow 20.0 --noise-model aLIGOZeroDetHighPower --output-filename BBH-IMRPhenomD-aLIGOZeroDetHighPower.hdf --convergence-threshold 25
+
+XML_SIZE=`ligolw_print BBH-IMRPhenomD-aLIGOZeroDetHighPower.xml -t sngl_inspiral | wc -l`
+HDF_SIZE=`h5ls BBH-IMRPhenomD-aLIGOZeroDetHighPower.hdf | grep "spin1z" | awk '{print $3}'`
+HDF_SIZE="${HDF_SIZE:1:3}"
+
+if ((XML_SIZE < 140 || XML_SIZE > 160)); then
+  echo "The XML bank is not the expected size " ${XML_SIZE}
+  exit 1
+fi
+
+if ((HDF_SIZE < 140 || HDF_SIZE > 160)); then
+  echo "The HDF bank is not the expected size " ${HDF_SIZE}
+  exit 1
+fi
+
```

Comment:
Can this be turned into a `pytest`-style test function?

Reply:
Possibly 🤷 . Any example of how this test would be done where I need to run the end executable and verify the output files in this way (there is some randomness, and I'm not convinced you get identical output every time).

... A general point is that I do think sbank needs more tests of individual functions/classes (which I'm not able to commit to doing right now), but we also need some end-to-end tests of the main executable(s), like this one, to make sure that everything is working.

Reply:
Ok, since `sbank` is a script, and not a fancy entry point, moving directly to a pytest script is not appropriate right now. However, I would like to try and incorporate the new test script into the coverage, which should be as simple as:

```patch
diff --git a/tools/test_sbank.sh b/tools/test_sbank.sh
index 6bfd570..1191839 100644
--- a/tools/test_sbank.sh
+++ b/tools/test_sbank.sh
@@ -1,8 +1,12 @@
 set -e
+
+PREFIX=$(python -c "import sys; print(sys.prefix)")
+PYTHON="python -m coverage run --append"
+
 # Run sbank to generate XML and HDF banks
-sbank --approximant IMRPhenomD --aligned-spin --mass1-min 15.0 --mass1-max 25.0 --spin1-min 0.0 --spin1-max 0.5 --match-min 0.97 --flow 20.0 --noise-model aLIGOZeroDetHighPower --output-filename BBH-IMRPhenomD-aLIGOZeroDetHighPower.xml --convergence-threshold 25
+${PYTHON} ${PREFIX}/bin/sbank --approximant IMRPhenomD --aligned-spin --mass1-min 15.0 --mass1-max 25.0 --spin1-min 0.0 --spin1-max 0.5 --match-min 0.97 --flow 20.0 --noise-model aLIGOZeroDetHighPower --output-filename BBH-IMRPhenomD-aLIGOZeroDetHighPower.xml --convergence-threshold 25

-sbank --approximant IMRPhenomD --aligned-spin --mass1-min 15.0 --mass1-max 25.0 --spin1-min 0.0 --spin1-max 0.5 --match-min 0.97 --flow 20.0 --noise-model aLIGOZeroDetHighPower --output-filename BBH-IMRPhenomD-aLIGOZeroDetHighPower.hdf --convergence-threshold 25
+${PYTHON} ${PREFIX}/bin/sbank --approximant IMRPhenomD --aligned-spin --mass1-min 15.0 --mass1-max 25.0 --spin1-min 0.0 --spin1-max 0.5 --match-min 0.97 --flow 20.0 --noise-model aLIGOZeroDetHighPower --output-filename BBH-IMRPhenomD-aLIGOZeroDetHighPower.hdf --convergence-threshold 25

 XML_SIZE=`ligolw_print BBH-IMRPhenomD-aLIGOZeroDetHighPower.xml -t sngl_inspiral | wc -l`
 HDF_SIZE=`h5ls BBH-IMRPhenomD-aLIGOZeroDetHighPower.hdf | grep "spin1z" | awk '{print $3}'`
```

Reply:
Okay, I've made this change as suggested.